### DISCUSSION TITLE
How to use a local version of clangd?

### DISCUSSION THREAD

**Discussion by R-Goc** [LazyVim/LazyVim#4168] [Q&A]
Hello, I am currently using clangd from the LazyExtras extension. However, I would like to use one of the clangd versions from my system, preferably the one in PATH. The reason for that is that I can't get the clangd extension to work with MS headers, which someone else suggested that using the system clangd solves, and also I would like to use the clangd from the Intel LLVM tree, to be able to parse SYCL headers. How should I do that?

**Comment by fikovnik** [comment]
I was having the same issue. Not sure if there is a prettier solution, but I ended up with the following:

In `~/.config/nvim/lua/plugins/clangd.lua`:

```lua
return {
  "neovim/nvim-lspconfig",
  opts = {
    servers = {
      clangd = {
        mason = vim.fn.executable("clangd") == 0,
      },
    },
  },
}
```

Basically, if `mason == false` it won't be configured by mason (IIUC).
